get_ndvi_data.R

```diff
@@ -1,39 +1,152 @@
 
-years = 2001:2015
-for (year in years){
-  coord_data_year <- coord_data
-  coord_data_year$start.date <- rep(as.Date(paste(year, "-06-01", sep="")), nrow(coord_data_year))
-  coord_data_year$end.date <- rep(as.Date(paste(year, "-06-30", sep="")), nrow(coord_data_year))
-  coord_date_data <- rbind(coord_date_data, coord_data_year)
+  bbs_data <- try(read.csv("data/bbs_data.csv"))
+  if(class(bbs_data)=='try-error'){stop("Can't load bbs_data.csv inside process_gimms_ndvi_bbs()")}
+  route_locations <- unique(dplyr::select(bbs_data, site_id, long, lat))
+  coordinates(route_locations) <- c("long", "lat")
+  
+  gimms_files=list.files(gimms_folder, full.names = TRUE)
+  #hdr files are created from some of the gimms processing that we don't want to
+  #use here.
+  gimms_files=gimms_files[!grepl('hdr', gimms_files)]
+  
+  gimms_ndvi_bbs=data.frame()
+  for(file_path in gimms_files){
+    gimms_ndvi_bbs=extract_gimms_data(file_path, route_locations) %>%
+      bind_rows(gimms_ndvi_bbs)
+  }
+  
+  
+  
+  return(gimms_ndvi_bbs)
 }
 
-if (length(list.files("./data/modisdata/100kmsq/")) > 0){
-  unaquired_coord_data <- UpdateSubsets(LoadDat = coord_date_data, StartDate = TRUE,
-                                        Dir = "./data/modisdata/100kmsq/")
-} else {
-  unaquired_coord_data <- coord_date_data
+#################################################
+#Filter the quality flags that we don't want.
+#Average the bimonthly values into 1 value per month
+#and add in -99 for na
+#From the GIMMS readme:
+#FLAG = 7 (missing data)
+#FLAG = 6 (NDVI retrieved from average seasonal profile, possibly snow)
+#FLAG = 5 (NDVI retrieved from average seasonal profile)
+#FLAG = 4 (NDVI retrieved from spline interpolation, possibly snow)
+#FLAG = 3 (NDVI retrieved from spline interpolation)
+#FLAG = 2 (Good value)
+#FLAG = 1 (Good value)
+#################################################
+filter_gimms_data=function(df){
+  df=df %>%
+    filter(flag<=3) %>%
+    group_by(site_id, year, month) %>%
+    summarize(ndvi=mean(ndvi)) %>%
+    ungroup() %>%
+    right_join( expand(df, site_id, month, year)) %>%
+    replace_na(list(ndvi=-99))
+  return(df)
 }
 
-MODISSubsets(LoadDat = unaquired_coord_data, Products = "MOD13Q1",
-             Bands = c("250m_16_days_NDVI", "250m_16_days_pixel_reliability"),
-             Size = c(5,5), SaveDir = "./data/modisdata/100kmsq/", StartDate = TRUE)
+
+#################################################
+#Get the GIMMS AVHRR ndvi bi-monthly time series for every bbs site.
+#Pulling from the sqlite DB or extracting it from raw gimms data
+#################################################
+get_bbs_gimms_ndvi = function(){
+  if('gimms_ndvi_bbs_data' %in% src_tbls(database)){
+    return(collect(tbl(database, sql('SELECT * from gimms_ndvi_bbs_data'))))
+  } else {
+    print('Gimms NDVI bbs data not found, processing from scratch')
+    
+    file_status=check_if_gimms_files_present()
+    if(!isTRUE(file_status)){
```

Comment:
And then... `if(!is.null(files_to_download)` or maybe clearer `if(length(files_to_download) > 0)`